### ISSUE TITLE
Conversations. switching between questions. & input validation

### ISSUE THREAD

**Issue by mrnuser** [grammyjs/examples#28]
Hi.
I have a quite popular task for which I haven't found any examples.

### 1. You need to switch between questions.
System:

```
question 1: How much?
user: 100
question 2: ure good?
user: Yes.
question 3: Want some apples?
user: back
question 2: ure good?
user: back
question 1: How much?
```

as you can see - the user switched back 2 questions and the bot asked them again. Of course, after the user answers, the bot will move on in the script and the user can also go back.

### 2. Input validation.
_Criterion for user's answer to question 1: parseInt(ctx.message.text)._
System:

```
question 1: How much?
user: asdjajsdja
question 1: How much?
user: asaasd
question 1: How much?
user: 100
question 2: ure good?
```

the bot did not accept the answer until it met the criterion.
Of course, this should work in conjunction with point 1 (switching between questions).

**Comment by LWJerri** [comment]
Hello! As I wrote to you in chat, you should get a basic understanding of writing code, we told you all morning yesterday how to do everything, but instead you spent the whole day trying to make a loop. What you described above is already in @grammyjs/conversations, like I told you above - learn basic programming and then come back to us =)

Ref: https://grammy.dev/plugins/conversations.html#functions-and-recursion

**Comment by KnorpelSenf** [comment]
No. We definitely need more examples with conversations. One of them should cover this issue. We can close it when we add the examples.

**Comment by KnorpelSenf** [comment]
I am currently working on the second major version of the conversations plugin. We should wait with this issue until it is released.

**Comment by LWJerri** [comment]
@KnorpelSenf, is this still actual rn?

**Comment by KnorpelSenf** [comment]
Conversations 2.0 was released so we can now add more examples. That being said, the plugin's API and the docs have improved significantly, so the relevance of this issue is lower than it was at the time it was opened.

**Comment by KnorpelSenf** [comment]
I actually think there are enough examples in the docs now. If somebody disagrees, we can get back to this later.
